### PR TITLE
Proba distribs

### PR DESCRIPTION
**Pull request by jtlap**
cauchy and exponential with attributes and docs

### CODE REVIEW

File: include/eve/function/probas_attributes.hpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
+//==================================================================================================
```
Comment (jfalcou) [inline include/eve/function/probas_attributes.hpp:1]: attributes.hpp ??

File: include/eve/function/probas_attributes.hpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  EVE_MAKE_CALLABLE(var_, var);   ;          // variance
+  EVE_MAKE_CALLABLE(entropy_, entropy);      // entropy
+  EVE_MAKE_CALLABLE(fisher_, fisher);        // fisher information
+  EVE_MAKE_CALLABLE(skewness_, skewness);    // skewness
+  EVE_MAKE_CALLABLE(kurtosis_, kurtosis);    // kurtosis
+  EVE_MAKE_CALLABLE(scale_, scale);          // scale
+
+  namespace detail
+  {
+
```
Comment (jfalcou) [inline include/eve/function/probas_attributes.hpp:49]: spurrious blank

File: include/eve/module/real/proba/function/regular/generic/cauchy.hpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      return if_else(is_ltz(x) || x > one(as(x)), allbits, tmp);
+    }
+
+    //////////////////////////////////////////////////////
+    /// parameters
+    template<floating_value T>
+    EVE_FORCEINLINE  auto parameters_(EVE_SUPPORTS(cpu_)
+                                     , cauchy<T> const &expo) noexcept
+    {
+      return std::make_tuple(expo.m,  expo.s);
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/cauchy.hpp:146]: I think this make_tuple is unneeded as one can use structured binding to retrieve m and s properly

File: include/eve/module/real/proba/function/regular/generic/cauchy.hpp [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    /// median
+    template<floating_value T>
+    EVE_FORCEINLINE  auto median_(EVE_SUPPORTS(cpu_)
+                                 , cauchy<T> const &expo) noexcept
+    {
+      return expo.m;
+    }
+
+    //////////////////////////////////////////////////////
+    /// scale
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/cauchy.hpp:159]: How many distributions have a scale?
This one attributes looks very specific.

File: include/eve/module/real/proba/function/regular/generic/cauchy.hpp [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+      return expo.m;
+    }
+
+    //////////////////////////////////////////////////////
+    /// entropy
+    template<floating_value T>
+    EVE_FORCEINLINE  auto entropy_(EVE_SUPPORTS(cpu_)
+                                  , cauchy<T> const & expo) noexcept
+    {
+      return log(4*pi(as(expo.s))*expo.s);
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/cauchy.hpp:181]: eve::log

File: include/eve/module/real/proba/function/regular/generic/exponential.hpp [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+      return if_else(t < expo.lambda, expo.lambda/(expo.lambda-t),  allbits);
+    }
+
+    //////////////////////////////////////////////////////
+    /// parameters
+    template<floating_value T>
+    EVE_FORCEINLINE  auto parameters_(EVE_SUPPORTS(cpu_)
+                                     , exponential<T> const &expo) noexcept
+    {
+      return expo.lambda;
```
Comment (jfalcou) [inline include/eve/module/real/proba/function/regular/generic/exponential.hpp:136]: Same, we should just return the structure and let structured binding do its job.
There we have one case returning a tuple, one not. This is quite confusing.

File: include/eve/module/real/proba/function/regular/generic/cauchy.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/real/proba/function/regular/generic/cauchy.hpp:159]: les distro qui n'ont pas de variance

**Pull request merged**